web/src/components/organisms/Dashboard/hooks.ts

```diff
@@ -71,20 +77,36 @@ export default (workspaceId?: string) => {
     [createWorkspaceMutation, setCurrentWorkspace, refetch, navigate]
   );
 
-  const handleModalClose = useCallback(() => {
-    setModalShown(false);
-  }, []);
+  const handleWorkspaceModalClose = useCallback(
+    (r?: boolean) => {
+      setWorkspaceModalShown(false);
+      if (r) {
+        refetch();
+      }
+    },
+    [refetch]
+  );
```

Comment:
I think I commented on code like this before. If the r param is necessary, and if it means to refetch, just name it refetch for clarity.

Reply:
But actually, refetch shouldn't be inside this callback, should it? 
Seems like this function is doing two seemingly unrelated things: closing the workspace modal, and checking whether or not we want to refetch the Me query. I feel like the creation function should handle this, since it deals with sending the mutation, and refetching is related to that. 

I see in the WorkspaceCreationModal and ProjectCreationModal are refetching even on cancelling the modal. Is that necessary?

Reply:
No, code was inspired from re:earth, I don't need to refetch so I removed the param